src/java.base/share/classes/jdk/internal/classfile/impl/SplitConstantPool.java

```diff
@@ -370,6 +374,26 @@ public AbstractPoolEntry.ClassEntryImpl classEntry(Utf8Entry nameEntry) {
         return e == null ? internalAdd(new AbstractPoolEntry.ClassEntryImpl(this, size, ne)) : e;
     }
 
+    @Override
+    public ClassEntry classEntry(ClassDesc classDesc) {
+        if (classDesc == ConstantDescs.CD_Object) {
```

Comment:
What causes the slow lookup of Object CE? If it's because that hashing needs to compute the substring first, I recommend changing the hashing algorithm if that's the case.

Reply:
Right, I'll take this specific shortcut back and will think about more generic performance improvement.